Formula/c/cahute.rb

```diff
@@ -0,0 +1,26 @@
+class Cahute < Formula
+  desc "Library and set of utilities to interact with Casio calculators"
+  homepage "https://cahuteproject.org/"
+  url "https://ftp.cahuteproject.org/releases/cahute-0.2.tar.gz"
+  sha256 "70e7f12747bee8b5be7c29ecbb0c447b7c54708c3cb31fb141396776dbbe9eee"
+  license "CECILL-2.1"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "libusb"
+  depends_on "sdl2"
+
+  def install
+    system "cmake", "-S", ".", "-B", "build", *std_cmake_args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+    bin.install "build/p7os"
+  end
+
+  test do
+    system "#{bin}/p7", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I added a new test case for the library (compiling a test program from the documentation), but I'm not sure what the best way to go about testing the utilities is. If a calculator is connected, they would succeed, but if one isn't connected, they would fail. How would I turn that into a reproducible test?

Reply:
Just assume none is connected. In most runs it will not be